dotnet/src/Connectors/Connectors.Memory.AzureCosmosDBMongoDB/AzureCosmosDBMongoDBVectorStoreRecordCollection.cs

```diff
@@ -435,5 +514,115 @@ private static Dictionary<string, string> GetStoragePropertyNames(
         return storagePropertyNames;
     }
 
+    /// <summary>
+    /// Get vector property to use for a search by using the storage name for the field name from options
+    /// if available, and falling back to the first vector property in <typeparamref name="TRecord"/> if not.
+    /// </summary>
+    /// <param name="optionsVectorFieldName">The vector field name provided via options.</param>
+    /// <exception cref="InvalidOperationException">Thrown if the provided field name is not a valid field name.</exception>
+    private VectorStoreRecordVectorProperty? GetVectorPropertyForSearch(string? optionsVectorFieldName)
+    {
+        // If vector property name is provided in options, try to find it in schema or throw an exception.
+        if (!string.IsNullOrWhiteSpace(optionsVectorFieldName))
+        {
+            var vectorDataModelPropertyNameExists = this._dataModelPropertyNames
+                .TryGetValue(optionsVectorFieldName!, out var vectorDataModelPropertyName);
```

Comment:
As mentioned in chat.  I didn't support storage names here, since they can differ by store implementation, so if a user uses them it's potentially breaking out of the abstraction.  So I didn't think it was that useful, and it can also have sideaffects.

Reply:
Yes, I already removed that logic to rely only on data model property name.